packages/block-editor/src/components/gradients/use-gradient.js

```diff
@@ -67,7 +65,15 @@ export function __experimentalUseGradient( {
 } = {} ) {
 	const { clientId } = useBlockEditContext();
 
-	const gradients = useSetting( 'color.gradients' ) || EMPTY_ARRAY;
+	const { gradients: gradientsPerOrigin } = useSetting( 'color' ) || {};
```

Comment:
Some of the properties, such as `useSetting( 'color.custom' )` have a special handling only takes effect when accessing them directly, and doing `useSetting( 'color' )` introduces some bugs. See https://github.com/WordPress/gutenberg/issues/37094

Until that's addressed we're reverting some of these changes at https://github.com/WordPress/gutenberg/pull/36900